### PR TITLE
feat(recovery): Add functionality of `do it later` in inline recovery key

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We want to be able to dismiss this promo

## This pull request

- Adds functionalit to the do it later button
- Sets a localstorage value so that it wont appear again if the user clicked do it later
- Adds glean metric for button click

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-10080

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx [hunk 1]
```diff
@@ -79,8 +99,8 @@ export const InlineRecoveryKeySetup = ({
         return (
           <InlineRecoveryKeySetupCreate
             {...{ createRecoveryKeyHandler }}
```
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx:101]: Could use syntactic sugar here
```suggestion
            {...{ createRecoveryKeyHandler, doLaterHandler }}
```

File: packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx [hunk 2]
```diff
@@ -21,6 +22,25 @@ export const InlineRecoveryKeySetup = ({
...
+  const doLaterHandler = () => {
+    localStorage.setItem(Constants.DISABLE_PROMO_ACCOUNT_RECOVERY_KEY, 'true');
+    // We do a hard navigate because this page is still in the content server, this
+    // also keeps all query params so that correct metrics are emitted.
+    hardNavigate('/connect_another_device', {}, true);
+  };
+
+  // We don't show this promo if the user has dismissed it or dismissed it from
+  // settings
+  if (
```
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx:34]: Should this check happen at the previous step (i.e., in the `getSyncNavigate` function)?

File: packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx [hunk 3]
```diff
@@ -21,6 +22,25 @@ export const InlineRecoveryKeySetup = ({
   createRecoveryKeyHandler: () => Promise<void>;
   currentStep: number;
 } & RouteComponentProps) => {
+  const doLaterHandler = () => {
```
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx:25]: Clicking on the dismiss button from the inline flow should set a separate key from the the dismiss button in the settings promo. Dismissing one shouldn't dismiss both - we want to be able to show both and measure engagement/dismissal rates.

File: packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx [hunk 4]
```diff
@@ -79,8 +99,8 @@ export const InlineRecoveryKeySetup = ({
         return (
           <InlineRecoveryKeySetupCreate
             {...{ createRecoveryKeyHandler }}
-            // TODO with FXA-10079
-            doLaterHandler={() => Promise.resolve()}
+            doLaterHandler={doLaterHandler}
+            data-glean-id="inline_recovery_key_setup_create_do_it_later"
```
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx:103]: This should be directly in the "Do it later" button.

File: packages/fxa-settings/src/components/NotificationPromoBanner/index.tsx [hunk 5]
```diff
@@ -34,7 +39,7 @@ const NotificationPromoBanner = ({
   // This value is used to animate the banner out of view when the user dismisses it.
   const [isClosing, setIsClosing] = useState(false);
 
-  const notificationBannerClosedLocalStorageKey = `__fxa_storage.fxa_disable_notification_banner.${dismissKey}`;
+  const notificationBannerClosedLocalStorageKey = PromoKeys[dismissKey];
```
Comment (vpomerleau) [inline packages/fxa-settings/src/components/NotificationPromoBanner/index.tsx:42]: Noted below, but this key should be distinct from the key used for the "do it later" button in the inline flow.

File: packages/fxa-settings/src/pages/InlineRecoveryKeySetup/container.tsx [hunk 6]
```diff
@@ -12,7 +12,7 @@ export const InlineRecoveryKeySetupContainer = ({
 }: {
   integration: Integration;
 } & RouteComponentProps) => {
-  const [currentStep, setCurrentStep] = useState<number>(1);
+  const [currentStep] = useState<number>(1);
```
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/InlineRecoveryKeySetup/container.tsx:15]: The set state function will be needed to make this functional in `FXA-10079`, might as well leave it in ;)

**Comment by vbudhram** [comment]
@vpomerleau I've update this PR to make it have a separate key from the "do it later" key.

File: packages/fxa-settings/src/components/NotificationPromoBanner/index.tsx [hunk 7]
```diff
@@ -34,7 +40,7 @@ const NotificationPromoBanner = ({
...
```
Comment (vpomerleau) [inline packages/fxa-settings/src/components/NotificationPromoBanner/index.tsx:43]: 👌 nice that this is extensible for future banners

**Review by vpomerleau — APPROVED** [review]
WFM, thanks for separating the local storage keys for the 2 promos!

**Pull request merged**